File: tps/TP1/PalindromoC/palindromo.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

// função que verifica se uma string é um palíndromo
bool validPalindromo(const char *txt) {
    int i = 0;
    int j = strlen(txt) - 1;

    while (i < j) {
        if (txt[i] != txt[j]) {
            return false;
        }
        i++;
        j--;
    }

    return true;
}

int main() {
    char texto[1000];
    bool flag = false;

    while (!flag) {
        // utilização do fgets por ineficácia do scanf
        if (fgets(texto, sizeof(texto), stdin) != NULL) {
            // remove o caractere de nova linha
            texto[strcspn(texto, "\n")] = '\0';

            // verifica se é necessário encerrar o programa
            if (strlen(texto) == 3 && texto[0] == 'F' && texto[1] == 'I' && texto[2] == 'M') {
                flag = true;
            } else {
                // chamada do método
                if (validPalindromo(texto)) {
                    printf("SIM\n");
                } else {
                    printf("NAO\n");
                }
            }
        }
    }


    return 0;
}

File: tps/tp1/q01/Palindromo.java
import java.util.Scanner;

public class Palindromo {

    // Método que checa se a string é um palíndromo
    public static boolean validPalindromo(String txt) {
        for (int i = 0, j = txt.length() - 1; i < j; i++, j--) {
            if (txt.charAt(i) != txt.charAt(j)) {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String texto = "";
        boolean flag = false;

        for (int i = 0; flag != true; i++) {
            texto = scanner.nextLine();

            if (texto.length() == 3 && texto.charAt(0) == 'F' && texto.charAt(1) == 'I' && texto.charAt(2) == 'M') {
                flag = true;
            } else {
                if (validPalindromo(texto)) {
                    System.out.println("SIM");
                } else {
                    System.out.println("NAO");
                }
            }
        }

        scanner.close();
    }
}

